### ISSUE TITLE
[python-package]What are the  benefits using softmax version in case of multiclass 

### ISSUE THREAD

**Issue by Cauchemare** [microsoft/LightGBM#6328]
### **Context**
the different raw scores can produce the same probabilities. 
For softmax, you should use all trees to get the probabilities after softmax.
BTW, there is a term in the second order gradient of softmax: https://github.com/Microsoft/LightGBM/blob/76102284d1997ef8b876cb40ae7c3e44b6ffbb66/src/objective/multiclass_objective.hpp#L86 
So, it actually cannot produce the extract same results with ova.

_Originally posted by @guolinke in https://github.com/microsoft/LightGBM/issues/1518#issuecomment-481074374_
            

### **Question**
There indeed exists  essential difference  between softmax version and general ova version in case of multiclass, my questions are what benefits using softmax compare with traditional ova version @guolinke
